std-bits/table/src/main/java/org/enso/table/data/index/MultiValueKey.java

```diff
@@ -49,8 +67,18 @@ public int hashCode() {
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
+
     MultiValueKey that = (MultiValueKey) o;
-    return hashCodeValue == that.hashCodeValue && Arrays.equals(values, that.values);
+
+    if (storage.length != that.storage.length) return false;
+    if (hashCodeValue != that.hashCodeValue) return false;
+    for (int i = 0; i < storage.length; i++) {
+      if (!Objects.equals(get(i), that.get(i))) {
```

Comment:
Does this work correctly for custom Enso types with a custom `==` definition?

Reply:
We use a custom object comparator for `compareTo`, but in `equals` we use the default Java equality.

Won't this cause problems? `compareTo` will compare `s\u0301` and `ś` as equal, but `equals` won't.

Reply:
Was meant to use the Object Comparator. Have amended.